src/bootstrap/src/core/download.rs

```diff
@@ -250,19 +257,67 @@ impl Config {
                             "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')",
                             url, tempfile.to_str().expect("invalid UTF-8 not supported with powershell downloads"),
                         ),
-                    ])).is_err() {
+                    ])).is_ok() {
                         return;
                     }
                     eprintln!("\nspurious failure, trying again");
                 }
             }
             if !help_on_error.is_empty() {
                 eprintln!("{help_on_error}");
+                Self::check_outdated(commit);
             }
             crate::exit!(1);
         }
     }
 
+    fn check_outdated(commit: &String) {
+        let build_date: String = String::from_utf8(
+            Command::new("git")
+                .arg("show")
+                .arg("-s")
+                .arg("--format=%ct")
+                .arg(commit)
+                .output()
+                .unwrap()
+                .stdout,
+        )
+        .unwrap_or("".to_string());
+        match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(n) => {
+                let replaced = build_date.trim();
+                let diff = n.as_secs() - replaced.parse::<u64>().unwrap();
+                if diff >= 165 * 24 * 60 * 60 {
+                    let build_date: String = String::from_utf8(
+                        Command::new("git")
+                            .arg("show")
+                            .arg("-s")
+                            .arg("--format=%ar")
+                            .arg(commit)
+                            .output()
+                            .unwrap()
+                            .stdout,
+                    )
+                    .unwrap_or("".to_string());
+                    if build_date.is_empty() {
+                        eprintln!(
+                            "NOTE: tried to download builds for {}, CI builds will be gone in 168 days",
+                            commit
+                        );
+                    } else {
+                        eprintln!(
+                            "NOTE: tried to download builds for {} (from {}), CI builds will be gone in 168 days",
+                            commit, build_date
+                        );
+                    }
+                    eprintln!("HELP: Consider updating your copy of the rust sources");
```

Comment:
This feels like a suggestion we could improve upon, but I guess it's OK for now. It seems like we still have a divergence between rustc components and LLVM components on how we compute the right commit to download, so there's not necessarily single guidance we can give here.